File: java/129. Sum Root to Leaf Numbers/Solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int total = 0;

    public int sumNumbers(TreeNode root) {
        dfs(root, 0);
        return total;
    }

    public void dfs(TreeNode node, int current) {
        if (node == null) return;

        current = current * 10 + node.val;
        //if leaf node, add into the total
        if (node.left == null && node.right == null) {
            total += current;
            return;
        }

        dfs(node.left, current);
        dfs(node.right, current);
    }

}

File: python/129. Sum Root to Leaf Numbers/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sumNumbers(self, root: Optional[TreeNode]) -> int:
        self.total = 0
        def dfs(node, current):
            if not node:
                return

            current = current * 10 + node.val

            if not node.left and not node.right:
                self.total += current
                return

            dfs(node.left, current)
            dfs(node.right, current)

        dfs(root, 0)
        return self.total
     